Sources/MacroToolkit/DeclGroup/AnyDeclGroup.swift

```diff
@@ -1,15 +1,18 @@
 import SwiftSyntax
 
-/// An enumeration representing different types of declaration groups (e.g., `struct`, `class`, `enum`, `actor`, `extension`).
-/// This enum conforms to `DeclGroupProtocol` and wraps specific declaration group types.
-public enum DeclGroup: DeclGroupProtocol {
+/// An enum that encapsulates various types of declaration groups (`struct`, `class`, `enum`, `actor`, `extension`)
+/// and provides a unified interface for interacting with them. This enum conforms to `AnyDeclGroupProtocol`,
+/// allowing access to common properties of declaration groups.
+public enum AnyDeclGroup: AnyDeclGroupProtocol {
```

Comment:
The goal of renaming to `AnyDeclGroup` was to align with Swift's conventions for type-erased types, like `AnyView`, `AnyHashable`, and `AnyCancellable`, making the distinction between concrete types and their type-erased counterparts clear and intuitive. This consistency with the standard library and clarity in differentiation helps in understanding the purpose of `AnyDeclGroup`. The aim was to allow `DeclGroupProtocol` to have a hard requirement on the type of its wrapped syntax while maintaining flexibility, and type erasure with `AnyDeclGroup` achieves this by holding any type conforming to `DeclGroupProtocol`. If we adopt `AnyDeclGroup`, similar updates should be applied to other root-level enums for consistency, but this is beyond the scope of the current PR and should be handled separately. For now, I've reverted to `DeclGroup` to match the existing naming convention, and if we decide to use the `Any` prefix, we should plan a broader refactor for consistency. I appreciate your feedback and am open to further discussion to find the best design approach.

Reply:
Yeah I can see why the `Any` prefix made sense cause of `DeclGroupSyntax` being a protocol for whatever weird reason. The reason that I don't have the `Any` prefix on any of the others (e.g. `Decl`) is because they're still concrete and not erasing any info (since they're enums), so I guess in this case it depends what's more important to devs; the fact that it's erasing the `_syntax` escape hatch, or the fact that it's not erasing the underlying decl group information.

Reply:
In-fact, `DeclGroup` can still conform to `RepresentableBySyntax`, but just with the underlying syntax type being `Syntax` instead (the main use would just be for using the syntax node in diagnostics, since all other useful decl group syntax info would unfortunately be erased)